ls1intum/Artemis
# `General` Data export for participations, submissions and results

# This PR contains **approved** database changes

### Checklist
#### General


#### Server
#### Client
#### Changes affecting Programming Exercises

### Motivation and Context


This is the first step implementing a data export for users of Artemis

### Description
In this PR I added the functionality to export personal data of exercise participations, submissions, and results
and  a basic user interface on the client side.
In the future this will become a scheduled nightly operation because it can potentially 
### Steps for Testing

Prerequisites:
- 1 Admin

1. Log in to Artemis
2. create participations, submissions, results for all exercise types
3.  Navigate to the privacy statement
4. Click on the data export button
5. Click on the request data export button (wait till the creation is finished)
6. Download the export and check that the information about your previously generated participations, submissions, results is in there



### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests


### Test Coverage





### Screenshots
